run/set_iptables.sh

```diff
@@ -3,92 +3,93 @@
 set +x
 set -e
 
-echo "INTERNAL_PORT=${INTERNAL_PORT:=9701}"
-echo "CLI_PORT=${CLI_PORT:=9702}"
-echo "CHAIN=${CHAIN:=DOCKER-USER}"
-echo "MAX_CONN=${MAX_CONN:=500}"
+source ./ip_tables_utils.sh
 
 usage() {
-  echo
-  echo "Usage:"
-  echo -n "INTERFACE=[your_network_interface] IP_FILE=[path_to_ip_addresses_file, defaults to first argument, env wins ties] "
-  echo -n "INTERNAL_PORT=[indy port, default 9701] CLI_PORT=[client port, default 9702] CHAIN=[iptables chain to edit, default DOCKER-USER]"
-  echo "$0 [IP_FILE]"
-  echo
-  echo "This script will add rules to your ip tables chain CHAIN to allow incoming connections on port INTERNAL_PORT"
-  echo "only from ips listed in the IP_FILE. It will also restrict the number of connections to port CLI_PORT to MAX_CONN."
-  echo
-  echo "The ip adresses file should contain the list of nodes"
-  echo "in your network. One ip address per line."
-  echo "The network interface should be the physical one used for incoming connections from the internet"
-  echo
-  echo "This script needs to be run as root/via sudo."
-  echo
+    echo
+    echo "Usage:"
+    echo -n "IP_FILE=[path_to_ip_addresses_file, defaults to first argument, env wins ties] "
+    echo -n "INTERNAL_PORT=[indy port, default 9701] CLI_PORT=[client port, default 9702] CHAIN=[iptables chain to edit, default DOCKER-USER]"
+    echo "$0 INTERFACE IP_FILE"
+    echo
+    echo "This script will add rules to your ip tables chain CHAIN to allow incoming connections on port INTERNAL_PORT"
+    echo "only from ips listed in the IP_FILE. It will also restrict the number of connections to port CLI_PORT to MAX_CONN."
+    echo
+    echo "The ip adresses file should contain the list of nodes"
+    echo "in your network. One ip address per line."
+    echo "The network interface INTERFACE should be the physical one used for incoming connections from the internet."
+    echo "Positional arguments can be given by environment variables instead. Env takes precedence."
+    echo
+    echo "This script needs to be run as root/via sudo."
+    echo
 }
 
-# skip existing rules to avoid duplicates
-add_new_rule() {
-  RULE="$@"
-
-  if iptables -C $RULE 2>/dev/null 1>&2; then
-    echo "[skip] $RULE already exists"
-  elif [[ "$RULE" == *"DROP"* ]] || [[ "$RULE" == *"RETURN"* ]]; then
-    iptables -A $RULE
-    echo "[ok] $RULE added to the end of the chain"
-  else
-    iptables -I $RULE
-    echo "[ok] $RULE added to the beginning of the chain"
-  fi
-}
-
-make_last_rule() {
-  RULE="$@"
-  while iptables -C $RULE 2>/dev/null 1>&2; do
-    iptables -D $RULE
-    echo "[ok] $RULE deleted"
-  done
-  iptables -A $RULE
-  echo "[ok] $RULE added to the end of the chain"
-}
+if ! ip_tables_installed; then
+    echo "Can not find/use iptables. Did you run this script with root priviledges?"
+    usage
+    exit 1
+fi
 
-echo "INTERFACE=${INTERFACE:=ens18}"
+echo "INTERFACE=${INTERFACE:=$1}"
```

Comment:
```suggestion
INTERFACE=$1
echo "INTERFACE=${INTERFACE:=ens18}"
```

Would allow us to overwrite and still have the default value

Reply:
`ens18` was our original default value. I am not sure if this is a sensible default for everybody in the frst place?